src/test/java/uk/gov/companieshouse/api/testdata/service/impl/TestDataServiceImplTest.java

```diff
@@ -94,6 +96,42 @@ void createCompanyDataDefaultSpec() throws DataException {
         assertEquals(AUTH_CODE, createdCompany.getAuthCode());
     }
 
+    @Test
+    void createCompanyDataScottishSpec() throws DataException {
+        CompanySpec spec = new CompanySpec();
+        spec.setJurisdiction(Jurisdiction.SCOTLAND);
+        CompanyProfile mockCompany = new CompanyProfile();
+        mockCompany.setCompanyNumber(COMPANY_NUMBER);
+
+        CompanyAuthCode mockAuthCode = new CompanyAuthCode();
+        mockAuthCode.setAuthCode(AUTH_CODE);
+
+        Appointment mockAppointment = new Appointment();
+        mockAppointment.setOfficerId(OFFICER_ID);
+        mockAppointment.setAppointmentId(APPOINTMENT_ID);
+
+        when(this.randomService.getNumber(anyInt())).thenReturn(Long.valueOf(COMPANY_NUMBER));
```

Comment:
Perhaps ensure we call it with the right parameter?

```suggestion
        when(this.randomService.getNumber(6).thenReturn(Long.valueOf(COMPANY_NUMBER));
```